packages/client/__tests__/AuthenticationPasswordReset.test.tsx

```diff
@@ -1332,5 +1392,53 @@ cases<{
                     },
                 ],
             },
+
+        'Goes up to new password screen, goes back to signing in screen and then submits again all forms to finally get authenticated':
+            {
+                target: 'Rendering home screen',
+                events: [
+                    {
+                        type: 'Type email',
+                        email: existingUser.email,
+                    },
+                    {
+                        type: 'Request password reset',
+                    },
+                    {
+                        type: 'Type on password reset code field',
+                        code: VALID_PASSWORD_RESET_CODE_LIST[0],
+                    },
+                    {
+                        type: 'Submit password reset token form',
+                    },
+                    {
+                        type: 'Type on new password field',
+                        newPassword: TEST_STRONG_PASSWORD[0],
+                    },
+                    {
+                        type: 'Go back to previous screen',
+                    },
+                    {
+                        type: 'Go back to previous screen',
+                    },
+                    {
+                        type: 'Request password reset',
+                    },
+                    {
+                        type: 'Type on password reset code field',
+                        code: VALID_PASSWORD_RESET_CODE_LIST[1],
```

Comment:
The previous VALID_PASSWORD_RESET_CODE_LIST[0] is well destroyed in any state machines context ?

Reply:
The `passwordResetCode` property will be reassigned, yes. Otherwise the test would not work as `@xstate/test` would have to go back to a state it has already visited! Which is not possible.